app/services/notify_service.rb

```diff
@@ -87,7 +87,7 @@ def response_to_recipient?
 
   def from_staff?
     sender = @notification.from_account
-    sender.local? && sender.user.present? && sender.user_role&.overrides?(@recipient.user_role) && @sender.user_role&.highlighted? && sender.user_role&.can?(*UserRole::Flags::CATEGORIES[:moderation])
+    sender.local? && sender.user.present? && sender.user_role&.overrides?(@recipient.user_role) && sender.user_role&.highlighted? && sender.user_role&.can?(*UserRole::Flags::CATEGORIES[:moderation].map(&:to_sym))
```

Comment:
Oh wow, so in 4.2 these were strings through `%w` and in 4.3 these are symbols via `%i` and that's what caused this change to break when backported?

Reply:
There were two issues, that, and a `@sender` instead of `sender`.

Reply:
oooh, ha, I missed that! Yikes. I do not envy the work of having to backport changes..

Reply:
Actually, why did we decide that the User's role needed to be highlighted to bypass? This kind renders the feature a bit useless for mods who aren't entirely public that they're a moderator.

(We really need #31081 and #27562 which would remove the need for this notification filter bypass feature)

Reply:
We did it so that “incognito” mods would not be outed. It was the reason behind the change.

Reply:
Mmm, okay, yeah.. I guess that makes sense.

I guess an operator/admin could always create a separate role of "Moderator" and "Moderator (private)", and make just the "Moderator" one highlighted?